cupy/lib/_routines_poly.py

```diff
@@ -147,6 +168,52 @@ def polymul(a1, a2):
     return cupy.convolve(a1, a2)
 
 
+_polyder_kernel = cupy._core.ElementwiseKernel(
+    'raw T p, int64 m',
+    'T z',
+    '''
+    T y = p[:-1] * p[_ind.size() - i - 1][1:];
+    if (m == 0) {
+        z = p;
+    } else {
+        z = polyder(y, m - 1);
+    }
```

Comment:
There is a problem with indexing, I think. Any hints about the same? Thanks!

Reply:
This string is embedded as a body of the CUDA `__global__` function, so Python-style array slicing is not available here.